sdk/anomalydetector/ai-anomaly-detector/review/ai-anomaly-detector.api.md

```diff
@@ -0,0 +1,120 @@
+## API Report File for "@azure/ai-anomaly-detector"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { KeyCredential } from '@azure/core-auth';
+import { OperationOptions } from '@azure/core-http';
+import { PipelineOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/identity';
+
+// @public
+export class AnomalyDetectorClient {
+    constructor(endpointUrl: string, credential: TokenCredential | KeyCredential, options?: AnomalyDetectorClientOptions);
+    detectChangePoint(body: DetectChangePointRequest, options?: OperationOptions): Promise<AnomalyDetectorClientDetectChangePointResponse>;
+    detectEntireSeries(body: DetectRequest, options?: OperationOptions): Promise<AnomalyDetectorClientDetectEntireResponse>;
+    detectLastPoint(body: DetectRequest, options?: OperationOptions): Promise<AnomalyDetectorClientDetectLastPointResponse>;
+    readonly endpointUrl: string;
+}
+
+// @public
+export type AnomalyDetectorClientDetectChangePointResponse = DetectChangePointResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: DetectChangePointResponse;
+    };
+};
+
+// @public
+export type AnomalyDetectorClientDetectEntireResponse = DetectEntireResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: DetectEntireResponse;
+    };
+};
+
+// @public
+export type AnomalyDetectorClientDetectLastPointResponse = DetectLastPointResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: DetectLastPointResponse;
+    };
+};
+
+// @public
+export interface AnomalyDetectorClientOptions extends PipelineOptions {
+}
+
+// @public (undocumented)
+export interface DetectChangePointRequest {
+    customInterval?: number;
+    granularity: TimeGranularity;
+    period?: number;
+    series: TimeSeriesPoint[];
+    stableTrendWindow?: number;
+    threshold?: number;
+}
+
+// @public (undocumented)
+export interface DetectChangePointResponse {
+    confidenceScores: number[];
+    isChangePoint: boolean[];
+    period: number;
+}
+
+// @public (undocumented)
+export interface DetectEntireResponse {
```

Comment:
since almost everything in this shape is an array, presumably because each index in the array is the value for a particular datapoint, would it make more sense to have something like

```ts
interface DetectEntireResponse {
    period: number;
    results: DetectEntireResponseItem[];
}

interface DetectEntireResponseItem {
    expectedValue: number;
    isAnomaly: boolean;
    isNegativeAnomaly: boolean;
    isPositiveAnomaly: boolean;
    lowerMargin: number;
    upperMargin: number;
}
```

or am I greatly misunderstanding what is being conveyed here?

Reply:
Yeah, this is something that Brian brought up and I agree, the feature crew decided to address this in the next preview across all languages.